source/crud/crud.rst

```diff
@@ -1655,7 +1655,7 @@ Find And Modify
 Update vs. Replace Validation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``update`` family of operations require that the update document parameter MUST have only atomic modifiers. In practice, this means that introspection needs to happen on that document to enforce this. However, it is enough to only check the first element in the document. If it begins with a ``$`` sign and the rest of the document's elements do not, the server will throw an error. Note that it is required that an update document have at least one atomic modifier.
+The ``update`` family of operations require that the update parameter, which can be either a document or a pipeline, MUST have only atomic modifiers. In practice, this means that introspection needs to happen on the document or pipeline to enforce this. However, it is enough to only check the first element in the update documents. If the element begins with a ``$`` sign and the rest of the document's elements do not, the server will throw an error. Note that it is required that the documents in the update parameter have at least one atomic modifier.
```

Comment:
I don't see any necessity for drivers to inspect the documents in a pipeline in the way that they do for single document update.  There's nothing it could be besides an aggregation pipeline.

Reply:
If I understand your comment correctly, we can trust that the server will require all elements in this BSON array (i.e. pipeline) to consist of atomic operators so there is no need for the driver to inspect elements and reject replacement documents as we do for `updateOne` and `updateMany` operations (and that purely to satisfy CRUD API and force users to use `replaceOne`). If so, I agree that relying on the server to raise an error makes sense.

Reply:
I have removed the comment.